core/modules/catalog/events.ts

```diff
@@ -110,3 +114,19 @@ export const onUserPricesRefreshed = async (store, router) => {
     }, { root: true })
   }
 }
+
+export const checkConfigurableParent = (store) => {
+  const parentProduct = store.getters['product/getParentProduct']
+  const currentProduct = store.getters['product/getCurrentProduct']
+  if (parentProduct && parentProduct.id !== currentProduct.id && config.products.preventConfigurableChildrenDirectAccess) {
+    Logger.log('Redirecting to parent, configurable product', parentProduct.sku)()
+    const parentUrl = formatProductLink(parentProduct, currentStoreView().storeCode)
+    AsyncDataLoader.push({
+      execute: async ({ context }) => {
+        if (context && !context.url.includes(parentUrl)) {
+          context.server.response.redirect(301, parentUrl)
+        }
+      }
+    })
+  }
+}
```

Comment:
It might be nice feature to add the configuration options of the child product as GET parameters. This would allow to have to correct filter options already active on PDP when user enters it.

What do you think?

Reply:
done, I've changed skus
```
    parentProduct.parentSku = parentProduct.sku
    parentProduct.sku = currentProduct.sku
```
So now it will redirect to proper configuration for parent